test/openst/02_create_user_wallet.js

```diff
@@ -0,0 +1,37 @@
+// 'use strict';
+
+const { assert } = require('chai');
+
+const { chainConfig, connection } = require('../shared');
+const OpenST = require('../../src/openst');
+
+describe('CreateUserWallet', async () => {
+  it('Creates user wallet', async () => {
+    const testAddresses = {
+      organization: '0x0000000000000000000000000000000000000001',
+      utilityToken: '0x0000000000000000000000000000000000000002',
+    };
+    const owner = connection.auxiliaryAccount.address;
+    const sessionKeys = [connection.auxiliaryAccount.address];
+    const openst = new OpenST(chainConfig, connection);
+    //  setupOpenst updates chainConfig with OpenST master copies and factory contracts.
+    await openst.setupOpenst(testAddresses.organization, testAddresses.utilityToken);
+
+    const {
+      tokenHolderProxy,
+      gnosisSafeProxy,
+      recoveryProxy,
+    } = await openst.createUserWallet(
+      testAddresses.utilityToken,
+      owner,
+      1,
+      sessionKeys,
+      [1000000000],
+      [10000000000],
+    );
+
+    assert.isNotNull(tokenHolderProxy, 'TokenHolder proxy address should not be null.');
```

Comment:
Does it make sense to confirm that the `owner` for the `tokenHolderProxy` is `gnosisSafeProxy` (and similar among/between these contracts to the extent other identities are known)?

cc: @pgev

Reply:
Isn't validating correct proxy address enough because setting owner for tokenHolderProxy is done inside UserWalletFactory contract. If you feel strongly on this, I can add the assertion. Let me know.

Reply:
The factory indeed sets the owner; it should be tested and caught at the unit tests in openst-contracts;

additionally we want to provide (later) a verifier executable that checks everything is setup correct, with canonical code; but here this check is not required.